iosApp/iosApp/Modules/Dashboard/DashboardPresenter.swift

```diff
@@ -147,11 +147,14 @@ private extension DefaultDashboardPresenter {
     }
     
     func didTapAction(_ idx: Int) {
-        guard let deepLink = DeepLink(identifier: actions[idx].deepLink) else { return }
-        if deepLink.identifier == DeepLink.themesList.identifier {
+        if (actions[idx] is Action.ConfirmMnemonic) {
+            wireframe.navigate(to: .mnemonicConfirmation)
+        }
+        if (actions[idx] is Action.Themes) {
             wireframe.navigate(to: .themePicker)
-        } else {
-            wireframe.navigate(to: .deepLink(deepLink))
+        }
+        if (actions[idx] is Action.ImprovementProposals) {
+            wireframe.navigate(to: .improvmentProposals)
         }
```

Comment:
Proffer explicit navigation via property name, instead of deep link string. Easier to follow without debugger